apps/web/lib/blog/mdx/components/mdx-components.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+
+import { cn } from "@/lib/utils";
+import DOMPurify from "dompurify";
+import type { MDXComponents } from "mdx/types";
+import Image from "next/image";
+import {
+  ComponentPropsWithoutRef,
+  ReactNode,
+  useDeferredValue,
+  useMemo,
+} from "react";
+import { MDX_STYLES } from "../constants";
+import {
+  extractLanguageFromClassName,
+  extractTextContent,
+  highlightCode,
+  looksLikeCode,
+} from "../utils/code-utils";
+import { CodeBlock } from "./code-block";
+
+type CodeProps = ComponentPropsWithoutRef<"code"> & {
+  className?: string;
+  children?: ReactNode;
+};
+
+function Code({ className, children, ...props }: CodeProps) {
+  const language = extractLanguageFromClassName(className);
```

Comment:
this fees like a lot of magic, is there no way to just pass along a `language=` prop? or is this extracting it from the MDX-generated classnames? feels very fragile